Wrappers/Python/cil/optimisation/algorithms/PDHG.py

```diff
@@ -17,55 +17,156 @@
 
 from cil.optimisation.algorithms import Algorithm
 import warnings
-
+import numpy as np
 
 
 class PDHG(Algorithm):
-    r'''Primal Dual Hybrid Gradient
+
+    r"""Primal Dual Hybrid Gradient (PDHG) algorithm, see :cite:`CP2011`, :cite:`EZXC2010`.
+
+    Parameters
+    ----------
+    f : Function
+        A convex function with a "simple" proximal method of its conjugate.
+    g : Function
+        A convex function with a "simple" proximal.
+    operator : LinearOperator    
+        A Linear Operator.
+    sigma : positive :obj:`float`, optional, default=None
+        Step size for the dual problem.
+    tau : positive :obj:`float`, optional, default=None
+        Step size for the primal problem.
+    initial : DataContainer, optional, default=None
+        Initial point for the PDHG algorithm.
+    use_axbpy: :obj:`bool`, optional, default=True
+        Computes a*x + b*y in C.
+    gamma_g : positive :obj:`float`, optional, default=None
+        Strongly convex constant if the function g is strongly convex. Allows primal acceleration of the PDHG algorithm.
+    gamma_fconj : positive :obj:`float`, optional, default=None
+        Strongly convex constant if the convex conjugate of f is strongly convex. Allows dual acceleration of the PDHG algorithm.
+
+    **kwargs:
+        Keyward arguments used from the base class :class:`Algorithm`.    
     
-    Problem: 
+        max_iteration : :obj:`int`, optional, default=0
+            Maximum number of iterations of the PDHG.
+        update_objective_interval : :obj:`int`, optional, default=1
+            Evaluates objectives, e.g., primal/dual/primal-dual gap every ``update_objective_interval``.
+
+
+    Example 
+    -------
+    Total variation denoising with with PDHG.  
+
+    .. math:: \min_{x\in X} \|u - b\|^{2} + \alpha\|\nabla u\|_{2,1}
+
+    >>> data = dataexample.CAMERA.get()
```

Comment:
This example could easily get out of date as it references lots of code from other functions. I think examples need to be as succinct as possible, preferably just an example of using the class, not a CIL full example running code.

Reply:
I agree, but configuring this algorithm requires some additional knowledge and a good example would be more approriate. Could we make this example less specific? 

```
f is an instance of a Function, e.g. MixedL21Norm
g is an instance of a Function, e.g. L2NormSquared
operator is an instance of a LinearOperator, e.g. GradientOperator
```

Then we could reference one of the CIL-Demos.